Update build settings for 1.14.2 forge

jadar opened the pull request:
### Overview
I am taking a look at what it takes to update the mod to 1.14.2. These are the changes I needed to import the project into IntelliJ. 

### Changes
- Update forge/MC version for 1.14.2.
- Add `mods.toml` file.

Commit:
Update build settings for 1.14.2 forge

- Update forge/MC version for 1.14.2.
- Add `mods.toml` file.

Kobata commented on build.gradle:
```diff
@@ -167,14 +167,14 @@ if(hasProperty('mod_core_plugin')) {
     options.debugOptions.debugLevel = 'source,lines,vars'
 }
 
+/*
```
While this stuff isn't important for locally building (in fact it shouldn't really do anything unless you've set up the configuration correctly), it is *quite* important for the CI/release build process so just disabling it isn't really a good idea.

Kobata commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -0,0 +1,52 @@
+# This is an example mods.toml file. It contains the data relating to the loading mods.
```
You probably don't need all the example text, especially stuff that's very easily outdated like the comments on what the current versions are.

The short description of each field is fine, though.

Kobata commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -0,0 +1,52 @@
+# A URL for the "homepage" for this mod, displayed in the mod UI
+#displayURL="http://example.com/" #optional
+# A file name (in the root of the mod JAR) containing a logo for display
+#logoFile="examplemod.png" #optional
```
This one and the next few are in the current mcmod.info file I believe.

Commit:
Uncomment curseforge deployment code.

Tested reimport, seems to still work for that.

Commit:
Update missing information in mods.toml

Commit:
Re-enable cursegradle plugin

jadar commented:
Let me know if there are any other changes that you would like to see. I think I addressed all your comments, so far. I clicked resolve but I'm not sure if that marks everything as "fixed". I haven't used Github's code review process much before.

safetypins commented:
I don't have a lot of experience with Java, but I'd also like to help if I can.
